### PR TITLE
feat: Adds new tokens to `Skeleton`

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to apply new tokens to `Skeleton` and its related components, based on the new [Theming structure](https://github.com/vtex-sites/base.store/pull/407): new scoped tokens linked to global ones.

It also uses FastStore UI's `Skeleton` component and renames `SkeletonElement` to `Skeleton` for a better code understanding/legibility.

## How does it work?

This PR uses local variables to parameterize the `Skeleton` and its related components' properties, then connects these scoped tokens to the global ones.

## How to test it?

These changes should not affect the skeletons' behavior. It can be tested through browser developer tools, applying a network throttling to simulate, for instance, a slow 3g connection.

https://user-images.githubusercontent.com/15722605/179318781-53dd772a-2e68-44f1-b721-5af49232ac26.mov

## Checklist

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [x] PR description

### CODE REVIEW

File: src/components/skeletons/ProductCardSkeleton/product-card-skeleton.module.scss [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  --fs-product-card-skeleton-min-width-mobile               : 14vw;
+  --fs-product-card-skeleton-min-width-desktop              : 12rem;
+  --fs-product-card-skeleton-height                         : 100%;
+  --fs-product-card-skeleton-padding                        : var(--fs-spacing-1) var(--fs-spacing-1) var(--fs-spacing-2);
+  --fs-product-card-skeleton-border                         : var(--fs-border-width) solid transparent;
+  --fs-product-card-skeleton-border-radius                  : var(--fs-border-radius);
+
+  // Image
+  --fs-product-card-skeleton-image-height-mobile            : 45vw;
+  --fs-product-card-skeleton-image-height-desktop           : #{rem(195px)};
```
Comment (ArthurTriis1) [inline src/components/skeletons/ProductCardSkeleton/product-card-skeleton.module.scss:18]: I think it doesn't work, in other parts we are converting manually
```suggestion
  --fs-product-card-skeleton-image-height-desktop           : 12.188rem;
```

**Review by ArthurTriis1 — COMMENTED** [review]
Good! I left just one comment. But we won't add a story for Skeleton?

File: src/components/skeletons/ProductTilesSkeleton/ProductTileSkeleton/product-tile-skeleton.module.scss [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+.fs-product-tile-skeleton {
+  // --------------------------------------------------------
+  // Design Tokens for Product Tile Skeleton
+  // --------------------------------------------------------
+
+  --fs-product-tile-skeleton-width                              : 100%;
+  --fs-product-tile-skeleton-min-width                          : var(--fs-product-tile-skeleton-width);
+  --fs-product-tile-skeleton-max-width                          : var(--fs-product-tile-skeleton-width);
+  --fs-product-tile-skeleton-height                             : var(--fs-product-tile-skeleton-width);
+  --fs-product-tile-skeleton-padding                            : 0;
```
Comment (eduardoformiga) [inline src/components/skeletons/ProductTilesSkeleton/ProductTileSkeleton/product-tile-skeleton.module.scss:12]: Not sure if they would be tokens, doesn't seem to change often.

File: src/components/skeletons/Shimmer/shimmer.module.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
+.fs-shimmer {
+  // --------------------------------------------------------
+  // Design Tokens for Shimmer
+  // --------------------------------------------------------
+
+  // Default properties
+  --fs-shimmer-wrapper-width                : 100vw;
+  --fs-shimmer-wrapper-height               : 100%;
+  --fs-shimmer-wrapper-top                  : 0;
+  --fs-shimmer-wrapper-left                 : 0;
```
Comment (eduardoformiga) [inline src/components/skeletons/Shimmer/shimmer.module.scss:10]: the same here regarding tokens' existence.

File: src/components/skeletons/Skeletons.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+Instead of showing a blank page, displaying <code>Skeletons</code> while loading content makes the user feel like the store is more responsive and faster.
+
+Also by using <code>Skeletons</code> the store will be less likely to experience [Cumulative Layout Shilf (or CLS)](https://web.dev/i18n/en/cls/).
+
+---
+
+## Components
+
+<SectionList classes="sbdocs-skeletons">
```
Comment (eduardoformiga) [inline src/components/skeletons/Skeletons.stories.mdx:35]: It's up to you but what do you think about the grid version?
```suggestion
<SectionList classes="sbdocs-skeletons" grid="grid">
```
Screen Shot 2022-07-19 at 20 23 59

**Review by eduardoformiga — COMMENTED** [review]
Amazing Job, @lucasfp13! 
You also included `ProductShelf` 🎉 

Sometimes I'm in doubt if we need some tokens, especially the ones with `width` or `height` with `100%` or `0` values.

File: src/components/skeletons/ProductTilesSkeleton/ProductTileSkeleton/product-tile-skeleton.module.scss (hunk 2 above)
Comment (hellofanny) [reply src/components/skeletons/ProductTilesSkeleton/ProductTileSkeleton/product-tile-skeleton.module.scss:12]: I think we don't need `--fs-product-tile-skeleton-min-width` and `--fs-product-tile-skeleton-max-width`  here

File: src/components/skeletons/Skeletons.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+# Skeletons
+
+This feature improves the store's experience and perceived performance through loading placeholders.
+
+</header>
+
+## Overview
+
+<code>Skeletons</code> are like an element's or section's wireframe. The wireframe
+is a placeholder for these elements/sections until the page finally loads.
```
Comment (hellofanny) [inline src/components/skeletons/Skeletons.stories.mdx:25]:
```suggestion
<code>Skeletons</code> are like an element's or section's wireframe. It's a placeholder that simulates the layout of these elements/sections while data is being loaded.
```

File: src/components/skeletons/Skeletons.stories.mdx (hunk 4 above)
Comment (hellofanny) [reply src/components/skeletons/Skeletons.stories.mdx:35]: Agreed! 👍

File: src/components/skeletons/Skeletons.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import Icon from 'src/components/ui/Icon'
+import ButtonLink from 'src/components/ui/Icon'
+
+<Meta title="Features/Skeletons/Overview" />
+
+<header>
+
+# Skeletons
+
+This feature improves the store's experience and perceived performance through loading placeholders.
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/skeletons/Skeletons.stories.mdx:18]: Nice question! I also wonder about it.

File: src/components/skeletons/FilterSkeleton/filter-skeleton.module.scss [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+  // Default properties
+  --fs-filter-skeleton-margin-top            : var(--fs-spacing-1);
+
+  // Title
+  --fs-filter-skeleton-title-max-width       : 30%;
+  --fs-filter-skeleton-title-margin-bottom   : var(--fs-spacing-2);
+
+  // Content
+  --fs-filter-skeleton-content-min-width     : 100%;
```
[the remark this answers is not in the document]
Comment (lucasfp13) [reply src/components/skeletons/FilterSkeleton/filter-skeleton.module.scss:16]: I don't have any scenario in mind, I will try to remove this and see if it affects somewhere

File: src/components/skeletons/Skeletons.stories.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (lucasfp13) [reply src/components/skeletons/Skeletons.stories.mdx:18]: I discussed this with @renatamottam, and it makes sense to be a feature for her, also I agree that it is ok so then we group all skeletons

File: src/components/skeletons/ProductCardSkeleton/product-card-skeleton.module.scss [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  // --------------------------------------------------------
+  // Design Tokens for Product Card Skeleton
+  // --------------------------------------------------------
+
+  // Default properties
+  --fs-product-card-skeleton-padding                        : var(--fs-spacing-1) var(--fs-spacing-1) var(--fs-spacing-2);
+  --fs-product-card-skeleton-border-width                   : var(--fs-border-width);
+  --fs-product-card-skeleton-border-color                   : transparent;
+  --fs-product-card-skeleton-border-style                   : solid;
+  --fs-product-card-skeleton-border-radius                  : var(--fs-border-radius);
```
Comment (renatamottam) [inline src/components/skeletons/ProductCardSkeleton/product-card-skeleton.module.scss:13]: Do we really need all these?

File: src/components/skeletons/FilterSkeleton/filter-skeleton.module.scss [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  [data-fs-filter-skeleton-content] {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    padding: var(--fs-filter-skeleton-content-padding);
+    overflow: hidden;
+    border-color: var(--fs-filter-skeleton-content-border-color);
+    border-style: style;
```
Comment (eduardoformiga) [inline src/components/skeletons/FilterSkeleton/filter-skeleton.module.scss:43]:
```suggestion
    border-style: solid;
```

or maybe group all props.
```css
border: var(--fs-filter-skeleton-content-border-width) solid var(--fs-filter-skeleton-content-border-color);
```

File: src/components/skeletons/FilterSkeleton/FilterSkeleton.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+# Filter Skeleton
+
+The loading placeholder for the <code>Filter</code> component used in PLP.
+
+</header>
+
+## Overview
+
+The `FilterSkeleton` component is part of [Skeletons](?path=/docs/features-skeletons-overview--page) feature.
```
Comment (hellofanny) [inline src/components/skeletons/FilterSkeleton/FilterSkeleton.stories.mdx:34]:
```suggestion
The `FilterSkeleton` component is part of [Skeletons](?path=/docs/features-skeletons-overview--usage) feature.
```

File: src/components/skeletons/ProductCardSkeleton/ProductCardSkeleton.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+
+# Product Card Skeleton
+
+The loading placeholder for the <code>ProductCard</code> used in several sections of the store.
+
+</header>
+
+## Overview
+
+The `ProductCardSkeleton` component is part of [Skeletons](?path=/docs/features-skeletons-overview--page) feature.
```
Comment (hellofanny) [inline src/components/skeletons/ProductCardSkeleton/ProductCardSkeleton.stories.mdx:42]:
```suggestion
The `ProductCardSkeleton` component is part of [Skeletons](?path=/docs/features-skeletons-overview--usage) feature.
```

File: src/components/skeletons/ProductCardSkeleton/product-card-skeleton.module.scss [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  // Badge
+  --fs-product-card-skeleton-badge-min-height          : var(--fs-spacing-5);
+  --fs-product-card-skeleton-badge-margin-top          : var(--fs-product-card-skeleton-text-margin-top);
+  --fs-product-card-skeleton-badge-max-width           : var(--fs-product-card-skeleton-content-height);
+  --fs-product-card-skeleton-badge-min-width-mobile    : 30%;
+  --fs-product-card-skeleton-badge-min-width-tablet    : var(--fs-product-card-skeleton-content-height);
+
+  // Sectioned
+  --fs-product-card-skeleton-sectioned-image-min-width : 13rem;
+  --fs-product-card-skeleton-sectioned-image-height    : var(--fs-product-card-skeleton-image-height-desktop);
```
Comment (hellofanny) [inline src/components/skeletons/ProductCardSkeleton/product-card-skeleton.module.scss:33]: I think  `--fs-product-card-skeleton-image-height-desktop`  variable is not defined before. Do you missed it or should it be `--fs-skeleton-image-height-desktop`? 🤔

**Review by hellofanny — APPROVED** [review]
Niceeee! I've left some notes, can you check it pls? 🌟

**Review by eduardoformiga — APPROVED** [review]
LGTM, after considering @hellofanny comments! 🎉

**Pull request merged**